src/overlays/actors/ovl_En_Lift_Nuts/z_en_lift_nuts.c

```diff
@@ -28,111 +62,930 @@ const ActorInit En_Lift_Nuts_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Lift_Nuts/func_80AEB8A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Lift_Nuts/func_80AEB934.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Lift_Nuts/func_80AEB974.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Lift_Nuts/func_80AEB9E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Lift_Nuts/func_80AEBB30.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Lift_Nuts/EnLiftNuts_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Lift_Nuts/func_80AEBC18.s")
+static Gfx* sDLists[] = {
+    object_dnt_DL_008290,
+    object_dnt_DL_001350,
+    object_dnt_DL_001420,
+    NULL,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Lift_Nuts/func_80AEBC90.s")
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
+    { 25, 75, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Lift_Nuts/EnLiftNuts_Draw.s")
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
+
+void func_80AE9A20(EnLiftNuts* this, PlayState* play) {
+    static s16 D_80AEBF60 = 0;
+    static s32 D_80AEBF64 = false;
+
+    if (!D_80AEBF64) {
+        func_800B6584(play, ACTOR_EN_GAMELUPY, &D_80AEBF60, sizeof(s16));
+        D_80AEBF64 = true;
+    }
+    this->ptr_1EC = &D_80AEBF60;
+}
+
+void func_80AE9A80(EnLiftNuts* this, PlayState* play) {
+    static s16 D_80AEBF68 = false;
+
+    if (!D_80AEBF68) {
+        func_800B6608(play, ACTOR_EN_GAMELUPY);
+        D_80AEBF68 = true;
+    }
+}
+
+s32 func_80AE9AC4(EnLiftNuts* this, s32 arg1) {
+    static s32 D_80AEBF6C = 0;
+
+    switch (arg1) {
+        case 0:
+            if (D_80AEBF6C == 0) {
+                return true;
+            }
+            break;
+        case 1:
+            if (D_80AEBF6C == 0) {
+                this->unk_34E = 1;
+                D_80AEBF6C = 1;
+                return true;
+            }
+            break;
+        case 2:
+            if (D_80AEBF6C == 1) {
+                this->unk_34E = 0;
+                D_80AEBF6C = 0;
+                return true;
+            }
+            break;
+        default:
+            break;
+    }
+    return false;
+}
+
+s32 func_80AE9B4C(s32 arg0, s32 arg1) {
+    static s32 D_80AEBF70 = 0;
+
+    if (arg0 == 0) {
+        if (D_80AEBF70 == arg1) {
+            return true;
+        }
+    } else if (arg0 == 1) {
+        D_80AEBF70 = arg1;
+        return true;
+    }
+    return false;
+}
+
+s32 func_80AE9B8C() {
+    s32 ret = false;
+
+    if (gSaveContext.save.weekEventReg[14] & 0x10) {
+        ret = true;
```

Comment:
```suggestion
    s32 ret = 0;

    if (gSaveContext.save.weekEventReg[14] & 0x10) {
        ret = 1;
```
Doesn't really look like a boolean.